### PR TITLE
Harvest: turn on default HttpAvailabilityStrategy

### LINKED ISSUE #21781: Source Harvest: Enable AvailabilityStrategy

### PR DESCRIPTION
**Pull request by grubberr**
Signed-off-by: Sergey Chvalyuk <grubberr@gmail.com>

## What
Issue: https://github.com/airbytehq/airbyte/issues/22254 https://github.com/airbytehq/airbyte/issues/21781
Turn on default HttpAvailabilityStrategy - it works ok for this connector

## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
Are there any breaking changes? What is the end result perceived by the user? If yes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.

## Pre-merge Checklist
Expand the relevant checklist and delete the others.

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- [ ] Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [ ] Secrets in the connector's spec are annotated with `airbyte_secret`
- [ ] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated
    - [ ] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [ ] Changelog updated in `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
- [ ] PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests)

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing
- [ ] New Connector version released on Dockerhub and connector version bumped by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)

</details>

### CODE REVIEW

**Review by lazebnyi — APPROVED** [review]

**Review by erohmensing — APPROVED** [review]

File: airbyte-integrations/connectors/source-harvest/integration_tests/expected_records.jsonl [hunk 1]
```diff
@@ -102,12 +102,12 @@
...
 {"stream":"project_budget","data":{"project_id":28671448,"project_name":"Monthly Retainer","client_id":10748671,"client_name":"[SAMPLE] Client B","budget_is_monthly":true,"budget_by":"project_cost","is_active":true,"budget":21910,"budget_spent":0,"budget_remaining":21910},"emitted_at":1672846398700}
 {"stream":"project_budget","data":{"project_id":28671447,"project_name":"Time & Materials Project","client_id":10748671,"client_name":"[SAMPLE] Client B","budget_is_monthly":false,"budget_by":"person","is_active":true,"budget":150,"budget_spent":166.21,"budget_remaining":-16.21},"emitted_at":1672846398700}
+{"stream":"time_team","data":{"user_id":3758382,"user_name":"[SAMPLE] Hiromi Hourglass","is_contractor":false,"weekly_capacity":144000,"avatar_url":"https://cache.harvestapp.com/assets/avatars/sample_avatar_2.png","total_hours":162.23,"billable_hours":129.77,"currency":"USD","billable_amount":11227.5,"from":"20210101","to":"20220101"},"emitted_at":1675795652626}
```
Comment (erohmensing) [inline airbyte-integrations/connectors/source-harvest/integration_tests/expected_records.jsonl:109]: Why did these change? Looks like harvest wasn't failing nightlies, and I wouldn't expect this change to affect the expected records

Comment (grubberr) [reply airbyte-integrations/connectors/source-harvest/integration_tests/expected_records.jsonl:109]: They constantly change
if @darynaishchenko last PR https://github.com/airbytehq/airbyte/pull/22406 works ok
I will keep her last version

**Comment by octavia-approvington** [comment]
After a careful ML study, <br/> we think this looks okay.  <br/> imagine code being okay

**Review by octavia-approvington — APPROVED** [review]

**Comment by cgardens** [comment]
@grubberr this is not passing the build. i'm going to revert it.

**Comment by cgardens** [comment]
@grubberr you must have a green build before merging.

**Pull request merged**